lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -868,11 +868,30 @@ internal class EvaluatingCompiler(
 
         val fieldThunks = fields.map {
             val (nameExpr, valueExpr) = it
+            if (nameExpr is Literal && !nameExpr.ionValue.isText) {
+                // Compile time error. Fail before the StructFieldThunks are evaluated.
+                err("Found struct field to be of type ${nameExpr.ionValue.type}",
```

Comment:
nit: similar to comment in `ErrorCode.kt`, error message should say struct field *name*/*key*. Same w/ on L887